ext/hal/ti/simplelink/Kconfig

```diff
@@ -11,7 +11,7 @@ config SIMPLELINK_HOST_DRIVER
 	default n
 	depends on HAS_CC3220SDK
 	depends on MULTITHREADING
-	depends on NEWLIB_LIBC
+	select NEWLIB_LIBC
```

Comment:
Why do we need newlib here?

Reply:
Well, actually, the SimpleLink host driver fails to compile without newlib. 
Some missing symbols: uint_least8_t (from stdint.h), and then when sockets are being used, it needs fcntl.h.
It's the same issue with Zephyr sockets (all samples are compiling with NEWLIB_LIBC).
Given cc3220 has 1Mb flash and 256K SRAM, it's not a huge issue - but with other platforms, it might be.

Reply:
> Well, actually, the SimpleLink host driver fails to compile without newlib.
Some missing symbols: ...

@GAnthony: This becomes a recurring problem (e.g. just had it with looking into cypress psoc6 work started). We need to submit tickets for these and fix whenever possible.